packit_service/worker/build/build_helper.py

```diff
@@ -91,6 +97,47 @@ def local_project(self) -> LocalProject:
             )
         return self._local_project
 
+    @property
+    def is_gitlab_instance(self) -> bool:
+        if self._is_gitlab_instance is None:
+            self._is_gitlab_instance = isinstance(self.project, GitlabProject)
+
+        return self._is_gitlab_instance
+
+    @property
+    def pr_id(self) -> Optional[int]:
+        if self._pr_id is None:
+            self._pr_id = self.metadata.pr_id
+        return self._pr_id
+
+    @property
+    def is_reporting_allowed(self) -> bool:
+        username = self.project.service.user.get_username()
+        if self._is_reporting_allowed is None:
+            self._is_reporting_allowed = self.base_project.can_merge_pr(username)
+        return self._is_reporting_allowed
+
+    @property
+    def base_project(self) -> GitProject:
+        """
+        Getting the source project info from PR,
+        In case of build events we loose the source info.
+        """
+        if self._base_project is None:
+            if self.pr_id:
+                self._base_project = self.project.get_pr(
+                    pr_id=self.pr_id
+                ).source_project
+            else:
+                self._base_project = self.project
+        return self._base_project
+
+    def request_project_access(self) -> None:
+        try:
+            self.base_project.request_access()
+        except GitlabAPIException:
+            logger.info("Access already requested")
```

Comment:
Is there something in the exception instance or exception string representation what can we use to filter only those that mean `Access already requested`?